samples/transfer-workflow-compensation/src/main/java/com/example/transfer/application/TransferWorkflow.java

```diff
@@ -36,160 +36,136 @@ public TransferWorkflow(ComponentClient componentClient) {
     this.componentClient = componentClient;
   }
 
-  // tag::definition[]
+
+  // tag::timeouts[]
+  // tag::recover-strategy[]
+  // tag::step-timeout[]
   @Override
-  public WorkflowDef<TransferState> definition() {
-    // tag::timeouts[]
-    // tag::recover-strategy[]
-    return workflow()
+  public WorkflowConfig configuration() {
+    return WorkflowConfig.builder()
       // end::recover-strategy[]
-      // end::timeouts[]
-      // tag::timeouts[]
-      .timeout(ofSeconds(5)) // <1>
+      // end::step-timeout[]
+      .workflowTimeout(ofSeconds(5)) // <1>
       .defaultStepTimeout(ofSeconds(2)) // <2>
       // end::timeouts[]
+      // tag::step-timeout[]
+      .stepConfig(TransferWorkflow::failoverHandlerStep, ofSeconds(1)) // <1>
+      // end::step-timeout[]
       // tag::recover-strategy[]
-      .failoverTo("failover-handler", maxRetries(0)) // <1>
-      .defaultStepRecoverStrategy(maxRetries(1).failoverTo("failover-handler")) // <2>
-      .addStep(withdrawStep())
-      .addStep(depositStep(), maxRetries(2).failoverTo("compensate-withdraw")) // <3>
-      // end::recover-strategy[]
-      .addStep(compensateWithdrawStep())
-      .addStep(waitForAcceptationStep())
-      .addStep(failoverHandlerStep());
+      .workflowRecovery(maxRetries(0).failoverTo(TransferWorkflow::failoverHandlerStep)) // <1>
+      .defaultStepRecovery(maxRetries(1).failoverTo(TransferWorkflow::failoverHandlerStep)) // <2>
+      .stepConfig(TransferWorkflow::depositStep, maxRetries(2).failoverTo(TransferWorkflow::compensateWithdrawStep)) // <3>
+      // tag::step-timeout[]
```

Comment:
We need to adapt the formatting here. It's creating a horizontal scroll bar. 

We can force some formatting using `// prettier-ignore`

See [prettier-formatter pr](https://github.com/akka/akka-sdk/pull/795/files#diff-9aba41592cfc054a3be4e22205f1515b6c730242cbbd7da9b3e4758e0664aa55R97).

Reply:
It's possible that we need to merge `main` into this feature branch in order to get the formatter.

Reply:
yeah I know, formatter is in the main branch, we can do a second round with formatting